### DISCUSSION TITLE
Open file in editor with sudo

### DISCUSSION THREAD

**Discussion by LaCocoRoco** [coder/code-server#5212] [Q&A]
Sometimes i need to edit a file with sudo (Example: Adding new reverse proxy into caddy file).
I am fine by editing the file with sudo nano in the terminal, but it would be very convenient to open it in the running code server editor.
One workaround is to spawn a new code-server instance as sudo on a different port with the working directory pointing to root.
Does someone know if there ia a way to open a file or folder with sudo in the editor?

**Comment by jsjoeio** [comment]
Hmm 🤔 I'm not sure to be honest. I found [this](https://stackoverflow.com/a/62423363/3015595) but I don't think it's much different from your current workaround

**Reply by jsjoeio** [reply]
cc @code-asher ideas?

**Reply by code-asher** [reply]
I am not aware of any more convenient workarounds but I think desktop VS Code handles this by asking for the password so maybe we can enable that behavior in code-server.

**Reply by LaCocoRoco** [reply]
@code-asher This would be a nice solution!

**Reply by LaCocoRoco** [reply]
@code-asher Hi, i thought to ask if there any new updates regarding this request?

**Reply by code-asher** [reply]
No updates; we have not looked into this.  If we can reproduce in Codespaces it could be nice to raise the issue upstream.

**Comment by eru123** [comment]
Hi there! Any luck on this?

I am connected to Ubuntu 22.04 via remote VSCode, I am trying to edit files for other user, which I don't want to change their file permission for me to edit their files normally.

As of now I use Vim when editing files and [Save as Root in Remote - SSH](https://marketplace.visualstudio.com/items?itemName=yy0931.save-as-root) which can be hassle sometimes.

**Comment by DasKraut37** [comment]
Also checking in on this… would be a great addition to code-server. 🙏

**Reply by Comzee** [reply]
Agree, it would make remote tweaks to service and OS files insanely easy!

**Comment by ThisIsQasim** [comment]
Incase anyone else is looking to do the same you can run `export EDITOR="code-server --wait"`  to achieve this. To make the settings persist, you should add it to your `~/.bash_profile`
